Looks like this fixed some validation:
```
:http-server-netty:compileTestGroovy FAILED
/home/runner/work/micronaut-core/micronaut-core/http-server-netty/src/test/groovy/io/micronaut/http/server/netty/websocket/QueryParamClientWebSocket.java:17: error: Parameter to @OnOpen must either be a URI variable, a WebSocketSession , the HttpRequest, or annotated with an HTTP binding annotation (such as @Header)
    public void onOpen(String dinner, WebSocketSession session, HttpRequest request) { // <3>
                              ^
Note: Creating bean classes for 37 type elements
1 error
startup failed:
Compilation failed; see the compiler error output for details.

1 error
```

Should we just remove the parameter? @yawkat

--- a/http-server-netty/src/test/groovy/io/micronaut/http/server/netty/websocket/QueryParamClientWebSocket.java
+++ b/http-server-netty/src/test/groovy/io/micronaut/http/server/netty/websocket/QueryParamClientWebSocket.java
@@ -1,6 +1,7 @@
 package io.micronaut.http.server.netty.websocket;
 
 import io.micronaut.http.HttpRequest;
+import io.micronaut.http.annotation.QueryValue;
 import io.micronaut.websocket.WebSocketSession;
 import io.micronaut.websocket.annotation.ClientWebSocket;
 import io.micronaut.websocket.annotation.OnMessage;
@@ -14,7 +15,7 @@ public class QueryParamClientWebSocket {
     private String dinner;
 
     @OnOpen
-    public void onOpen(String dinner, WebSocketSession session, HttpRequest request) { // <3>
+    public void onOpen(@QueryValue String dinner, WebSocketSession session, HttpRequest request) { // <3>
         this.session = session;
         this.request = request;
         this.dinner = dinner;

